bcml/cli.py

```diff
@@ -0,0 +1,184 @@
+                    "desc": meta["description"],
+                    "platform": "wiiu"
+                    if get_settings("wiiu")
+                    else "switch",
+                },
+            )
+    elif args.export:
+        # --export
+        install.export(args.export)
+
+    elif args.standalone:
+        # --standalone
+        bnp = args.standalone
+        with TempModContext():
+            install.install_mod(
+                bnp,
+                merge_now=True,
+                options={"options": {"texts": {"all_langs": True}}, "disable": []},
+            )
+            
+            out = f"{bnp.stem}.zip"
+            if out:
+                output = Path(out)
+                install.export(output, standalone=True)
+    
+    elif args.update:
+        # --update
+        update_file = args.update;
+        for i in get_installed_mods(disabled=True):
+            name = i.name.replace(" ", "").lower()
+            arg = update_file.name.rsplit(".")[0].replace(" ", "").lower()
+            if name == arg:
+                mod = i;
+
+        if (mod.path / "options.json").exists():
+            options = json.loads((mod.path / "options.json").read_text());
+        else:
+            options = {};
+
+        shutil.rmtree(mod.path)
+        with Pool(maxtasksperchild=500) as pool:
+            new_mod = install.install_mod(
+                Path(update_file),
+                insert_priority=mod.priority,
+                options=options,
+                pool=pool,
+                updated=True,
+            );
+
+        install.refresh_master_export()
+
+    elif args.remerge:
+        # --remerge
+        install.refresh_merges()
+        install.refresh_master_export()
+
+    elif args.install:
+        # --install
+        install.install_mod(
+            mod=args.install, 
+            insert_priority=args.priority if args.priority else 0, 
+            merge_now=True
+        )
+        install.refresh_master_export()
+
+    elif args.uninstall:
+        # --uninstall
+        for mod in util.get_installed_mods(disabled=True):
+            name = mod.name.replace(" ", "").lower()
+            arg = args.uninstall.replace(" ", "").lower()
+            if name.lower() == arg.lower():
+                install.uninstall_mod(mod=mod)
+    
+    elif args.uninstall_all:
+        for mod in get_installed_mods(disabled=True):
+            install.uninstall(mod=mod, wait_merge=True)
+            install.refresh_merges()
+            install.refresh_master_export()
+
+    elif args.backup:
+        # --backup
+        install.create_backup(args.backup)
+
+    elif args.restore:
+        # --restore
+        backup = list((util.get_storage_dir() / "backups").glob(f"{args.restore}*"))
+        if len(backup) <= 0:
+            raise ValueError("The backup does not exists!")
+
+        backup = backup[0]
+        install.restore_backup(backup)
+
+    else:
+        return
+
+    exit()
```

Comment:
2 things:
1) The `main()` loop *always* runs this function, so having this `exit()` here means bcml will never run the UI. There's multiple ways to solve this, but the one I prefer is to make a CLA that tells bcml to use CLI-mode, instead of UI mode. Then, this function can `exit()` all it wants

2) Please use trailing newlines in files. The big red ![image](https://user-images.githubusercontent.com/32864080/158030204-7a8eac3f-f368-429d-a4fd-1d86668bc243.png) displeases the machine spirits.

Reply:
In regards to 1., the function returns if none of the command line arguments are passed (aside from `--portable`), so main() will just skip it if there's nothing to do

Reply:
Ah. So it does, 3 lines above where I commented. Thanks, I had blinders on and missed that.